### PR TITLE
[Doc] Adding documentation for Rexe functions

### PR DESCRIPTION
**Pull request by srijan-sivakumar**
Fixes: #392

Signed-off-by: srijan-sivakumar <ssivakum@redhat.com>

### CODE REVIEW

**Comment by srijan-sivakumar** [comment]
To add abstract execute function details.

File: docs/BP/Ops/rexe.md [hunk 1]
```diff
@@ -0,0 +1,387 @@
+# Remote Execution Functions
+
+This document is specific to the functions dealing with the remote
+execution operations, what are their arguments, parameters and how to use
+them along with example.
+
+1. Rexe Class
+Why are we having a description about a class when we stated this document
+consists of functions falling under the category of remote execution
+operations ? Well, because the remote execution opeations can be performed
```
Comment (aujjwal-redhat) [inline docs/BP/Ops/rexe.md:10]: opeations => operations

File: docs/BP/Ops/rexe.md [hunk 2]
```diff
@@ -0,0 +1,387 @@
...
+* Return value: True if the command has been completed else False.
+* Purpose: After a request has been put for the execution of an async command,
+we need to check when it has finished execution and what is its current running
+status.
+* Where can I use it: This method can be used in ops as well as in test cases
+if the need is to check whether the async command has finished execution or
+not.
+* Example:
+
+-> Whether we use is inside an ops or directly in a test case ( in case of
```
Comment (aujjwal-redhat) [inline docs/BP/Ops/rexe.md:158]: is => it

File: docs/BP/Ops/rexe.md [hunk 3]
```diff
@@ -0,0 +1,387 @@
...
+* Purpose: The purpose of this method is to collect the results after the
+asynchronous command has been executed.
+* Where can I use it: We can use this along with the `execute_command_async`,
+as result collection is not handled by the `execute_command_async` hence the
+requirement for a separate function to obtain the results and pass it on after
+the execution has been finished. This can be used both at ops as well as
+test case level.
+* Example:
+
+-> Whether we use is inside an ops or directly in a test case ( in case of
```
Comment (aujjwal-redhat) [inline docs/BP/Ops/rexe.md:204]: is =>it

File: docs/BP/Ops/rexe.md [hunk 4]
```diff
@@ -0,0 +1,387 @@
...
+* Purpose: This method logically encapsultes the operations of two other
+methods, `check_async_command_status` and `collect_async_result`. So, 
+technically this is equivalent to telling the framework that I want to wait
+till I receive a response for the said asynchronous command which I had
+requested for.
+* Where can I use it: Again, ops or test cases, go crazy ( according to
+requirement ).
+* Example:
+
+-> Whether we use is inside an ops or directly in a test case ( in case of
```
Comment (aujjwal-redhat) [inline docs/BP/Ops/rexe.md:253]: is => it

File: docs/BP/Ops/rexe.md [hunk 5]
```diff
@@ -0,0 +1,387 @@
...
+method to use whereas if they want to directly handle the validations instead
+of leaving the command status validation to a try-except validation layer,
+this can be used.
+But that said, this function was mainly envisaged to be used by gluster related
+ops functions. these ops functions will usually create the cmd but use the
+`execute_command_multinode` to actually execute those commands in the 
+remote servers.
+* Example:
+
+-> Whether we use is inside an ops or directly in a test case ( in case of a
```
Comment (aujjwal-redhat) [inline docs/BP/Ops/rexe.md:325]: is => it

File: docs/BP/Ops/rexe.md [hunk 6]
```diff
@@ -0,0 +1,387 @@
...
+exceptions seen across the ops functions whenever the command execution
+had failed at the server.
+* When can I use it : This is the standard function which can be used if the
+user doesn't want to go ahead in the test case flow if the operation which had
+been requested has failed. If the `execute_command` sends back a response
+wherein the error code is assigned some value other than 0 ( which means
+success ), then this method will raise an exception.
+* Example:
+
+-> Whether we use is inside an ops or directly in a test case ( in case of a
```
Comment (aujjwal-redhat) [inline docs/BP/Ops/rexe.md:373]: is => it

**Review by aujjwal-redhat — CHANGES_REQUESTED** [review]

File: docs/BP/Ops/opsIndex.md [hunk 7]
```diff
@@ -3,4 +3,5 @@
 1. [Peer Ops]()
 2. [Brick Ops](./brick_ops.md)
 3. [Volume Ops]()
-4. [Profile Ops](./peer_ops.md)
\ No newline at end of file
+4. [Profile Ops](./peer_ops.md)
```
Comment (nishith-vihar) [inline docs/BP/Ops/opsIndex.md:6]: I think this should be profile_ops.md

**Review by aujjwal-redhat — APPROVED** [review]
LGTM

**Review by nishith-vihar — APPROVED** [review]
LGTM

**Pull request merged**